### ISSUE TITLE
8.10.10 firmware causes problems with kvm switch

### ISSUE THREAD

**Issue by AnsonT** [UltimateHackingKeyboard/agent#1559]
After upgrading from 8.10.9 to 8.10.10, UHK isn't recognized after switching between 2 PCs on a USB KVM switch.

Starting UHK Agent will restore functionality.

**Comment by mondalaci** [comment]
Are you sure your KVM played nicely with UHK firmware 8.10.9? Please double-check it.

**Comment by AnsonT** [comment]
Yes, I've been using 8.10.9 for while and upgrade yesterday. I'm trying to flash back to 8.10.9 to confirm, but not able to do so with the downloaded release from gitthub.  Do you have instructions on how to flash old versions of firmware?

**Comment by AnsonT** [comment]
OK, it appears that it takes about 1/2 minute after switching for the keyboard to be recognized. That delay wasn't there with the older firmware.

**Comment by AnsonT** [comment]
FWIW, the trackpad module is recognized immediately after switching but the keyboard part is not.

**Comment by mondalaci** [comment]
@abcminiuser Any ideas which of [your changes](https://github.com/UltimateHackingKeyboard/firmware/compare/v8.10.9...v8.10.10) have caused this?

**Comment by abcminiuser** [comment]
Nothing's jumping out at me - we might have to make bisected test firmwares to try individually to narrow it down. There's only 16 commits, so that's a maximum of only 5 steps.

**Comment by AnsonT** [comment]
I'd be happy to test the various builds, just let me know where I can download the firmware. (Sorry, I'm not set up to build it myself at the moment)

**Comment by jfieber** [comment]
I too encountered a problem with 8.10.10 and a USB switch to switch between two Apple computers.  The keyboard is eventually recognized without manual intervention, but takes 5-10 seconds after switching. Reverted back to 8.10.9 and things are back to normal.

the USB switch: https://www.amazon.com/UGREEN-Selector-Computers-Peripheral-One-Button/dp/B01MXXQKGM
